File: ncs/rncslib/test/PropertiesTest.rb
#!/usr/bin/ruby -w

require 'test/unit'
require 'logger'
require 'ncs/Properties'

class PropertiesTest < Test::Unit::TestCase
	@@ClassName = 'PropertiesTest'
	def setup
		@properties = Properties.new
		ENV['NCS_HOME'] = @ncshome = File::dirname(__FILE__) + '/../../ncshome'
	end
	
	def test_load
		puts "----------test load method----------"
		@properties.load("#{@ncshome}/conf.init/ncs_sm5.0.properties")
		puts "#{@properties.size} key value pairs defined in the properties file"
		assert(@properties.size > 0, "the size should be more than 0")
		assert_equal(0, @properties.getint('ncs.option.compile_isl'))
		assert_equal(1, @properties.getint('ncs.option.compile_model'))
		assert_equal(1, @properties.getint('ncs.option.compile_test'))
		assert_equal('NONE', @properties.get('ncs.option.ignore_test'))
		assert_equal('openr6,rrm,motor6,rrm_motor6', @properties.get('ncs.test.suite'))
	end
	
	def test_setget
		puts "----------test set & get method----------"
		@properties.set('ncs.tmp.test', 'testsetget');
		assert_equal('testsetget', @properties.get('ncs.tmp.test'))
	end
	
	def test_dump
		@properties.load("#{@ncshome}/conf.init/ncs_sm5.0.properties")
		puts "----------test dump method----------"
		@properties.dump
	end
	
	def teardown
		@properties = nil
	end
end

File: ncs/pyncs/test/core/propertiestest.py
#!/usr/bin/env python
"""
properties test suite.
"""

import sys, os
import unittest

# adding current directory to path to make sure local modules can be imported
sys.path.insert(0, '.')

from ncs.core.properties import Properties

NCS_HOME=os.getcwd()+"/../ncshome"
os.environ["NCS_HOME"] = NCS_HOME

class PropertiesTest(unittest.TestCase):
    def setUp(self):
        self.props = Properties()
        #self.props.debugon()
        
    def test_load(self):
        self.props.load(NCS_HOME + "/conf.init/ncs_sm5.0.properties")
        self.assert_(self.props.size() > 0)
        self.props.dump()
    
    def test_get(self):
        self.props.load(NCS_HOME + "/conf.init/ncs_sm5.0.properties")
        self.assert_(self.props.get('ncs.test.compile_path'), '/vob/wibb_bts/msm/test/sm_test/cosim')
    
    def test_set(self):
        self.props.set('ncs.test.compile_path', '/vob/wibb_bts/msm/test/sm_test/cosim')
        self.assert_(self.props.get('ncs.test.compile_path'), '/vob/wibb_bts/msm/test/sm_test/cosim')
    
    def test_search(self):
        self.props.load(NCS_HOME + "/conf.init/ncs_sm5.0.properties")
        prjs = self.props.search('prj')
        self.assert_(len(prjs) == 5)
        for (key,value) in prjs.items():
            print("{0} = {1}".format(key,value))
    
    def test_getrange(self):
        self.props.load(NCS_HOME + "/conf.init/ncs_sm5.0.properties")
        cases = self.props.getrange('ncs.test.suite.openr6')
        self.assert_(len(cases) > 0)
        for case in cases: print(case)
        
        cases = self.props.getrange('ncs.test.suite.rrm_motor6')
        self.assert_(len(cases) == 1)
        self.assert_(cases[0] == 3150)
        
        cases = self.props.getrange('ncs.test.suite.motor6')
        self.assert_(len(cases) == 7)
        for case in cases: print(case)
        
if __name__ == "__main__":
    unittest.main()
    